src/main/cc/math/distributed_discrete_gaussian_noiser.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "math/distributed_discrete_gaussian_noiser.h"
+
+#include <random>
+
+#include "absl/random/bit_gen_ref.h"
+#include "absl/random/random.h"
+
+namespace wfa::math {
+
+DistributedDiscreteGaussianNoiser::DistributedDiscreteGaussianNoiser(
+    DistributedDiscreteGaussianNoiseComponentOptions options)
+    : options_(options) {}
+
+/**
+ * Generate a sample from a discrete Gaussian distribution with parameter sigma.
+ *
+ * This implementation is adapted from @pasin30055 implementation linked
+ * below:
+ * https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/common/noisers.py#L207
+ */
+absl::StatusOr<int64_t>
+DistributedDiscreteGaussianNoiser::GenerateNoiseComponent() {
+  absl::BitGen rnd;
+  double sigma_ = options_.sigma;
+  double sigma_sq = sigma_ * sigma_;
+  double t = std::floor(sigma_) + 1;
+  double p_geometric = 1 - exp(-1 / t);
+  if (p_geometric <= 0 || p_geometric >= 1) {
```

Comment:
I think this condition is just impossible due to how `p_geometric` is defined in the previous line. Maybe we can remove this. Wdyt?